### PR TITLE
Added pluralization system for lang translations

### PR DESCRIPTION
**Pull request by shubham1206agra**
### Fixed Issues

$ https://github.com/Expensify/App/issues/38614

### Tests

1. Navigate to a chat room or report that contains messages with pluralized text related to expense counts (e.g., "2 expenses, 1 scanning").
2. Verify that the pluralized text is displayed correctly based on the count (e.g., "1 expense" for count 1, "2 expenses, 1 scanning" for counts 2 expenses and 1 scanning receipt).
3. Change the app's language to a different supported language (e.g., Spanish) and verify that the pluralized text is correctly translated and displayed based on the counts.
4. Navigate to a workspace settings page that shows the "Selected" count for members (e.g., "2 selected").
5. Verify that the pluralized text is displayed correctly based on the count (e.g., "1 selected" for count 1, "5 selected" for count 5).
6. Change the app's language to a different supported language (e.g., Spanish) and verify that the pluralized text is correctly translated and displayed based on the count.
7. Navigate to a workspace settings page that displays options related to distance rates (e.g., "Delete rates", "Enable rates", "Disable rate", "Are you sure you want to delete these rates?").
8. Verify that the pluralized text is displayed correctly based on the count for each option.
9. Change the app's language to a different supported language (e.g., Spanish) and verify that the pluralized text is correctly translated and displayed based on the count for each option.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests

### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b1eb6855-f9d4-40f5-9f64-b2903c3f5b7b

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b18d8893-bddd-443d-b416-f3c31d7751bc

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/e774dd8b-709d-4459-926e-b3255d83808d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/cd028de3-0491-441a-a4d3-d61c4a9a2037

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/229b0caa-1daf-46a6-8138-a0dbdca54193

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/5712f04c-0d07-4213-8977-430b7fc3b4a7

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
> You can start testing this PR.

Sure, will start testing in an hour

File: src/languages/en.ts
Comment (jayeshmangwani) [reply src/languages/en.ts:3315]: IMO, this is not correct, it should be

```javascript
deleteRates: () => ({
    one: `Delete rate`,
    other: () => `Delete rates`,
}),
enableRates: () => ({
    one: `Enable rate`,
    other: () => `Enable rates`,
}),
disableRates: () => ({
    one: `Disable rate`,
    other: () => `Disable rates`,
}),
```

File: src/languages/en.ts
Comment (jayeshmangwani) [inline src/languages/en.ts:3326]:
```suggestion
            areYouSureDelete: () => ({
                one: 'Are you sure you want to delete this rate?',
                other: () => 'Are you sure you want to delete these rates?',
            }),
```

**Comment by jayeshmangwani** [comment]
@shubham1206agra Yes, I have tested it and suggested a few changes above. When running the app, I haven't faced any issues yet.

**Comment by jayeshmangwani** [comment]
@shubham1206agra If you feel that the PR is ready for verifying the TypeScript changes, then we can tag the TS expert to verify the type changes.

File: src/libs/Localize/index.ts
Comment (jayeshmangwani) [inline src/libs/Localize/index.ts:134]: Why are we setting the count default to 0 if there is a first element?

File: src/languages/en.ts
Comment (jayeshmangwani) [inline src/languages/en.ts:824]: If we pass the count value 0 for the confirmApprovalAllHoldAmount key, it does not return the text for zero(`This expense is on hold. Do you want to approve anyway?`); it goes to other case(`These expenses are on hold. Do you want to approve anyway?`).

Comment (jayeshmangwani) [reply src/languages/en.ts:824]: So, why have we added the zero string here if it will never be used?

File: src/libs/Localize/index.ts
Comment (jayeshmangwani) [reply src/libs/Localize/index.ts:134]: What I am trying to say is that we should not add the fallback to 0. If there is no count, then we can return the translated phrase, right?

File: src/languages/en.ts
Comment (jayeshmangwani) [reply src/languages/en.ts:824]: Ohhkay, then I think we can remove that unnecessary part. I will finish the testing today, and I still need to test cases for `src/libs/Localize/index.ts`, as this file is a little complicated and used throughout the app.

**Comment by jayeshmangwani** [comment]
@shubham1206agra The PR is working fine for me; I haven't found any issues. I think we can proceed with opening this PR for the type changes. WDYT ?

**Comment by jayeshmangwani** [comment]
@shubham1206agra Would you complete the checklist here?

**Comment by jayeshmangwani** [comment]
@blazejkustra Tagging you here for verifying 👀 the TypeScript changes on the PR

File: src/components/AddressForm.tsx
Comment (blazejkustra) [inline src/components/AddressForm.tsx:129]: This looks odd, why do you have to add undefied here? `zipFormat` is an optional argument:

```ts
zipFormat?: string
```

File: src/components/ArchivedReportFooter.tsx
Comment (blazejkustra) [inline src/components/ArchivedReportFooter.tsx:56]: `text` is now of type any 😢

File: src/components/ArchivedReportFooter.tsx
Comment (blazejkustra) [inline src/components/ArchivedReportFooter.tsx:57]: This looks less cleaner now, why is this change required?

File: src/components/Search/SearchPageHeader.tsx
Comment (blazejkustra) [inline src/components/Search/SearchPageHeader.tsx:225]: Having undefined just because you have to provide the third argument isn't ideal. I'm okay with leaving it like this but is there another way?

File: src/languages/types.ts
Comment (blazejkustra) [inline src/languages/types.ts:303]: Let's add an explanation what it does

**Review by blazejkustra — CHANGES_REQUESTED** [review]
Couple comments, looking really solid from TS perspective!

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/libs/Localize/index.ts
Comment (jayeshmangwani) [inline src/libs/Localize/index.ts:48]: Following the discussion from [this conversation](https://github.com/Expensify/App/pull/45892/files#r1696842532), I believe we need to handle the optional arguments within the PhraseParameters here. for not to add the undefined when argument is optional

```javascript
errors.zipPostCode = translate('privatePersonalDetails.error.incorrectZipFormat', undefined);
```

We could handle it like below, but I'm not sure if it's the best way. What are your suggestions on the approach @blazejkustra @shubham1206agra ?

```diff
- type PhraseParameters<T> = 
-    T extends (arg: infer A) => string ? [A] : 
-    T extends (arg: infer A) => PluralFormPhase ? [A, number] : 
-    never[];
+ type PhraseParameters<T> = 
+    T extends (arg?: infer A) => string ? [A?] :
+    T extends (arg: infer A) => PluralFormPhase ? [A, number]:
+    T extends (arg: infer A) => string ? [A] :
+    never[];
```

File: src/components/ArchivedReportFooter.tsx
Comment (blazejkustra) [reply src/components/ArchivedReportFooter.tsx]: Let's add  `: string` here, thanks!

Comment (jayeshmangwani) [reply src/components/ArchivedReportFooter.tsx]: @shubham1206agra @blazejkustra Alternatively, we can directly set the text value within the switch case as like below:

```javascript
const text =(()=> {
    switch (archiveReason) {
        case CONST.REPORT.ARCHIVE_REASON.ACCOUNT_CLOSED:
            return translate(`reportArchiveReasons.${archiveReason}`, {
                displayName: `<strong>${displayName}</strong>`,
            });
        case CONST.REPORT.ARCHIVE_REASON.ACCOUNT_MERGED:
            return translate(`reportArchiveReasons.${archiveReason}`, {
                displayName: `<strong>${displayName}</strong>`,
                oldDisplayName: `<strong>${oldDisplayName}</strong>`,
            });
        case CONST.REPORT.ARCHIVE_REASON.REMOVED_FROM_POLICY:
            return translate(`reportArchiveReasons.${archiveReason}`, {
                displayName: `<strong>${displayName}</strong>`,
                policyName: `<strong>${policyName}</strong>`,
                shouldUseYou: actorPersonalDetails?.accountID === getCurrentUserAccountID(),
            });
        case CONST.REPORT.ARCHIVE_REASON.POLICY_DELETED:
            return translate(`reportArchiveReasons.${archiveReason}`, {
                policyName: `<strong>${policyName}</strong>`,
            });
        default:
            return translate(`reportArchiveReasons.${archiveReason}`);
    }
})()
```

File: src/languages/en.ts
Comment (jayeshmangwani) [inline src/languages/en.ts:846]: Please remove zero key

File: src/languages/es.ts
Comment (jayeshmangwani) [inline src/languages/es.ts:839]: Remove this

File: src/components/ArchivedReportFooter.tsx
Comment (jayeshmangwani) [reply src/components/ArchivedReportFooter.tsx]: @shubham1206agra  why use let here? I don't see any logic in it since we aren't going to reassign the text.

File: src/components/Search/SearchPageHeader.tsx
Comment (iwiznia) [reply src/components/Search/SearchPageHeader.tsx:294]: > It will force every parameter block to use Object structure. Even though there is only one parameter.

Why is this bad? This was how it worked till it changed [here](https://github.com/Expensify/App/pull/29742/files#r1459284554). I have no idea why that change was made.

> It will require more extensive changes as it will require to change many translation parameter structures

What do you mean? Can you list some of them here?

>  It will remove the optional parameter typing (which will require us to pass an empty structure instead of nothing).

Why? I don't see why we could not still make the parameters param be optional.

Comment (iwiznia) [reply src/components/Search/SearchPageHeader.tsx:294]: > How will you detect if the type has optional members?

What do you mean?

Comment (jayeshmangwani) [reply src/components/Search/SearchPageHeader.tsx:294]: @shubham1206agra Any update here on moving ahead?

Comment (jayeshmangwani) [reply src/components/Search/SearchPageHeader.tsx:294]: >  I will try to do this in another follow-up PR.

Okay!
@iwiznia What are your thoughts on the other follow-up PR?

Comment (jayeshmangwani) [reply src/components/Search/SearchPageHeader.tsx:294]: > We have been waiting for this feature for over 6 months, we can wait longer and make it right, especially since if we change it later we will need to update all callers of the method using the count param.

Fair enough, @shubham1206agra We need to make the changes in this PR only, so please share any TypeScript errors you're encountering when adding the count to the phrase parameters. We can work together to resolve them and also get a second opinion from the TypeScript experts.

Comment (jayeshmangwani) [reply src/components/Search/SearchPageHeader.tsx:298]: @iwiznia @shubham1206agra I'm working on adding the count to the phrase parameters. Currently, I'm encountering 3-4 TypeScript errors and still need to test a few things. I'll push the code for review over the weekend.

Comment (jayeshmangwani) [reply src/components/Search/SearchPageHeader.tsx:298]: @shubham1206agra I've raised a draft PR https://github.com/Expensify/App/pull/47963 where I've attempted to pass the count parameter within the phrase parameters. However, I'm encountering a TypeScript error that I still need to resolve. I'd appreciate your thoughts on the PR and any feedback you might have on the changes.

**Comment by blazejkustra** [comment]
@shubham1206agra @jayeshmangwani @iwiznia What are the next steps here?

**Comment by jayeshmangwani** [comment]
> What are the next steps here?

@ZhenjaHorbach is working on this PR https://github.com/Expensify/App/pull/48229 to add the count parameter to the phrase parameter, and @shubham1206agra is reviewing it. If that PR works fine and we can overcome the existing TypeScript issues/warnings, we will either incorporate that code into this PR or close this one and open the other PR for review.

**Comment by blazejkustra** [comment]
Discussed the issue with @shubham1206agra on slack, I won't be able to help this week because I have other tasks with higher priority. 

I'll try to jump in on Monday and try to come with a solution to infinite recursion/union type that is too complex problems.

**Comment by blazejkustra** [comment]
Meanwhile, to make sure I understand correctly the thing we want to achieve is to enforce that the second argument is an object + limit to two arguments at max for translate function, right?

**Comment by blazejkustra** [comment]
Looking into it now 👀

**Comment by blazejkustra** [comment]
I started working on it [here](https://github.com/Expensify/App/pull/48914/files). It may be impossible to enforce that the second argument is an object unfortunately, could you elaborate why do we need it? @ZhenjaHorbach @shubham1206agra

**Comment by ZhenjaHorbach** [comment]
> I started working on it [here](https://github.com/Expensify/App/pull/48914/files). It may be impossible to enforce that the second argument is an object unfortunately, could you elaborate why do we need it? @ZhenjaHorbach @shubham1206agra

Could you check this PR ?
https://github.com/Expensify/App/pull/48229
I managed to make typing so that we could only pass an object 

And as far as I remember it was done according to this topic
https://github.com/Expensify/App/pull/45892#discussion_r1696847656

**Review by ZhenjaHorbach — APPROVED** [review]

**Review by iwiznia — DISMISSED** [review]
Going to approve since I posted this to see if we enable the eslint rule for it: https://expensify.slack.com/archives/C01GTK53T8Q/p1727441233067399

**Review by blazejkustra — APPROVED** [review]

File: src/languages/en.ts
Comment (iwiznia) [reply src/languages/en.ts]: Yeah:
> Going to approve since I posted this to see if we enable the eslint rule for it: https://expensify.slack.com/archives/C01GTK53T8Q/p1727441233067399

**Review by iwiznia — APPROVED** [review]

**Comment by iwiznia** [comment]
@jayeshmangwani can you re-review and tag me to merge once you are done, please?

**Comment by iwiznia** [comment]
Ah got it!

**Comment by iwiznia** [comment]
Same as in https://github.com/Expensify/App/pull/45892 this PR touches a lot of unrelated files and we don't want to fix existing eslint problems as part of it.

**Comment by kavimuru** [comment]
@jayeshmangwani @blazejkustra Could you help us with 1st 3 steps where do we see the following?

> 1. Navigate to a chat room or report that contains messages with pluralized text related to the maximum number of participants (e.g., "You've selected the maximum number (2) of participants.").
> 2. Verify that the pluralized text is displayed correctly based on the count (e.g., "You've selected the maximum number (2) of participants." for count 2, and "You've selected the maximum number (5) of participants." for count 5).
> 3. Change the app's language to a different supported language (e.g., Spanish) and verify that the pluralized text is correctly translated and displayed based on the count.

**Pull request merged**
